crates/nu-cli/src/shell/completer.rs

```diff
@@ -47,7 +48,12 @@ impl NuCompleter {
                         LocationType::Argument(_cmd, _arg_name) => {
                             // TODO use cmd and arg_name to narrow things down further
                             let path_completer = crate::completion::path::Completer::new();
-                            path_completer.complete(context, partial)
+                            let completed_paths = path_completer.complete(context, partial);
+                            if &line[..2] == "cd" {
```

Comment:
This only handles the case where the line is `cd ...`, but you'd miss out on the opportunity to complete `cd` in other parts of the line. For example:

```
some | line | pipeline; cd <TAB>
```

But you're in luck! See that `_cmd` in the match expression above? Remove the underscore, and what you're left with is an `Option<String>` that is the command name:

```rust
match cmd.as_deref().unwrap_or("") {
  "cd" => autocomplete_only_folders(completed_paths),
  _ => completed_paths,
}
```

Reply:
Thanks @thegedge!
I tried your solution, but this doesn't seem to work.
When I typed "cd <tab>", cd autocompleted not only folders but also files.

Reply:
Ah, that's probably because I need to fix this TODO: https://github.com/nushell/nushell/blob/4828a7cd296f501d3290439a22eda16a41f6a2ca/crates/nu-cli/src/completion/engine.rs#L96-L98 haha

Let's stick with what you have, and I can return and make this change 🙂